src/toast.tsx

```diff
@@ -263,6 +264,52 @@ export const Toast = React.forwardRef<ToastRef, ToastProps>(
       variant,
     });
 
+    const renderCloseButton = React.useMemo(() => {
+      if (!dismissible) {
+        return null;
+      }
+      if (close) {
+        return close;
+      }
+      if (typeof closeButton === 'boolean' && closeButton) {
```

Comment:
Isn't that typecheck redundant? Typescript should error on the user-land.

Reply:
I think i added that before so that it would only follow the closeBUtton logic if a boolean was supplied and not a value such as undefined, but that was before I returned `close` before the `closeButton`, so I can remove it now. Nice catch